### PR TITLE
docs: update stable release policy and add missing releases

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

- Adding extra 6 months to currently active releases.
- Supporting by default extra 6 months the new releases we make.
- Adding releases that were missing for callbacks and 08-wasm.

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
🙏🏻

File: RELEASES.md [hunk 1]
```diff
@@ -58,26 +58,36 @@ A release should not be finalized until the development team and the external co
 
 The beginning of a new major release series is marked by the release of a new major version. A major release series is comprised of all minor and patch releases made under the same major version number. The series continues to receive bug fixes (released as minor or patch releases) until it reaches end of life. The date when a major release series reaches end of life is determined by one of the two following methods:
 
-- If the next major release is made within the first 6 months, then the end of life date of the major release series is 1 year after its initial release.
-- If the next major release is made 6 months after the initial release, then the end of life date of the major release series is 6 months after the release date of the next major release.
+- If the next major release is made within the first 6 months, then the end of life date of the major release series is 18 months year after its initial release.
```
Comment (womensrights) [inline RELEASES.md:61]:
```suggestion
- If the next major release is made within the first 6 months, then the end of life date of the major release series is 18 months after its initial release.
```

**Review by womensrights — CHANGES_REQUESTED** [review]
thank you, just noticed one small typo to fix

File: RELEASES.md [hunk 2]
```diff
@@ -58,26 +58,36 @@ A release should not be finalized until the development team and the external co
...
-|`v0.1.x-ibc-go-v7.3.x`|March 17, 2024|
+|`v0.1.x+ibc-go-v7.3.x`|September 17, 2024|
+|`v0.1.x+ibc-go-v8.0.x`|May 10, 2025|
+
+### `08-wasm` light client proxy module
+
+|Release|End of Life Date|
+|-------|----------------|
+|`v0.1.0+ibc-go-v7.3.x-wasmvm-v1.5.x`|September 17, 2024|
+|`v0.2.x+ibc-go-v8.0.x-wasmvm-v1.5.x`|May 10, 2025|
```
Comment (colin-axner) [inline RELEASES.md:90]: should we add a note that we may drop pre v1.0 versions if we deem them unnecessary to maintain? Pre 1.0 versions are basically minor or major releases and really we are only looking to give long term support for major releases

**Review by colin-axner — APPROVED** [review]
ACK

File: RELEASES.md (hunk 2 above)
Comment (DimitrisJim) [reply RELEASES.md:90]: do we even want to explicitly talk about long term support? I feel like pre 1.0.0 releases are meant for rapid and potentially disruptive changes and long term support sounds antithetical to that and could potentially stifle it. Happy if a comment is at least added though, but wanted to raise this Q.

Comment (crodriguezvega) [reply RELEASES.md:90]: I added a note [here](https://github.com/cosmos/ibc-go/pull/5606/commits/30a7cb17d166c8648896ef90a1bd18e554216895).

**Pull request merged**